### PR TITLE
Revamping Tests

### PR DESCRIPTION
**Pull request by madflojo**
So I was thinking about what could be done to ensure that we can validate the core functionality with the various guest languages with every change.

This PR basically sets up the tests to run the same basic tests against a small, simple guest generated from various languages (AssemblyScript, Rust, & Go for now). The nice thing about this is that not only does it test each change against each language supported, it also can serve as an example to new users who want to get started.

Happy to make modifications if the direction doesn't jive with y'alls thoughts.

### CODE REVIEW

**Comment by madflojo** [comment]
@pkedy you get a chance to look at this PR? Any thoughts?

**Comment by pkedy** [comment]
@madflojo I’ll get to this soon.  Sorry for the delay!

**Comment by madflojo** [comment]
As an FYI, TinyGo v0.20 was released so I updated the Go wasm test module to the latest and greatest.

**Comment by pkedy** [comment]
@madflojo Awesome, thanks!  Just FYI I have looked at this PR but didn't have time to complete the review but it looks good. Things are busy ATM but I will get to it.

**Comment by madflojo** [comment]
@pkedy no worries at all, I maintain a few repos myself and totally get how busy things can get outside of those projects.

File: module_test.go [hunk 1]
```diff
@@ -1,62 +1,153 @@
...
+					t.Errorf("Expected error when calling unregistered function, got nil")
+				}
+			})
+
+		})
+	}
+}
+
+func TestModuleBadBytes(t *testing.T) {
+	b := []byte("Do not do this at home kids")
```
Comment (pkedy) [inline module_test.go:91]: 🤣

**Review by pkedy — APPROVED** [review]

**Pull request merged**